plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/StandardProjectWizard.java

```diff
@@ -58,10 +57,26 @@ public boolean performFinish() {
       status = new Status(Status.ERROR, "todo plugin ID", errorCode, ex.getMessage(), null);
     }
     
-    // todo if fail, call setErrorMessage()
+    showErrorMessageIfNecessary(status);
+    
     return status.isOK();
   }
 
+  
+  // visible for testing
+  void showErrorMessageIfNecessary(IStatus status) {
+    // if fail, display error message in wizard
+    if (status == Status.CANCEL_STATUS) {
+      page.setErrorMessage("User canceled project creation");
+    } else if (!status.isOK()) {
+      String message = "Failed to create project";
+      if (status.getMessage() != null && !status.getMessage().isEmpty()) {
+        message += ": " + status.getMessage();
+      }
+      page.setErrorMessage(message);
```

Comment:
I definitely want to show this to the user; i.e. not log it.

I believe this puts it in the wizard page's text area up at the top. Do you think that's preferable to opening a separate dialog?

I tried your suggestion of getCurrentPage(), or rather getWizardContainer().getCurrentPage(). Unfortunately that returns the wrong type, IWizardPage instead of WizardNewProjectCreationPage; and IWizardPage doesn't have the setErrorMessage method. I could cast it, but that defeats the point of using the current page. I think we may need to live with using the page field here (unless we pop up an extra dialog) and rely on the tests to catch this if at some point we add a second page, at which time we'll have to rework this method.